util/llbutil/secretprovider/aws_creds.go

```diff
@@ -0,0 +1,116 @@
+package secretprovider
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/moby/buildkit/session/secrets"
+	"github.com/pkg/errors"
+	"gopkg.in/ini.v1"
+)
+
+// AWSCredentialProvider can load AWS settings from the environment.
+type AWSCredentialProvider struct {
+	mu          sync.Mutex
+	config      *ini.File
+	credsConfig *ini.File
+}
+
+// NewAWSCredentialProvider creates and returns a credential provider for AWS.
+func NewAWSCredentialProvider() *AWSCredentialProvider {
+	return &AWSCredentialProvider{}
+}
+
+// GetSecret attempts to find an AWS secret from either the environment or a local config file.
+func (c *AWSCredentialProvider) GetSecret(ctx context.Context, name string) ([]byte, error) {
+
+	names := map[string]int{
```

Comment:
nit: use a `map[string]struct{}{` to make it more clear this is being used as a set rather than a map.

Reply:
I often use `struct{}` for larger maps considering the byte savings. But in this case, I think `int` is a little prettier than this:

```
	names := map[string]struct{}{
		"AWS_ACCESS_KEY_ID":     struct{}{},
		"AWS_SECRET_ACCESS_KEY": struct{}{},
		"AWS_SESSION_TOKEN":     struct{}{},
		"AWS_DEFAULT_REGION":    struct{}{},
	}
```

Update: my editor reminded me that the `struct{}` is redundant here.

```
	names := map[string]struct{}{
		"AWS_ACCESS_KEY_ID":     {},
		"AWS_SECRET_ACCESS_KEY": {},
		"AWS_SESSION_TOKEN":     {},
		"AWS_DEFAULT_REGION":    {},
	}
```

That's nicer. 

![image](https://github.com/earthly/earthly/assets/332408/d42aef4c-6328-43a3-9b54-3b98dc122aed)